cpp-client/deephaven/dhcore/src/ticking/immer_table_state.cc

```diff
@@ -216,77 +219,138 @@ std::shared_ptr<RowSequence> MyTable::GetRowSequence() const {
   return rb.Build();
 }
 
-struct FlexVectorFromTypeMaker final {
-  template<typename T>
-  void operator()() {
-    if constexpr(DeephavenTraits<T>::kIsNumeric) {
-      result_ = std::make_unique<NumericAbstractFlexVector<T>>();
-    } else {
-      result_ = std::make_unique<GenericAbstractFlexVector<T>>();
-    }
+std::unique_ptr<AbstractFlexVectorBase> MakeFlexVectorFromType(const ElementType &element_type) {
+  if (element_type.ListDepth() != 0) {
+    const char *message = "Don't know how to make flex vectors with element types that are lists";
+    throw std::runtime_error(DEEPHAVEN_LOCATION_STR(message));
   }
 
-  std::unique_ptr<AbstractFlexVectorBase> result_;
-};
+  switch (element_type.Id()) {
+    case ElementTypeId::kChar: {
+      return std::make_unique<NumericAbstractFlexVector<char16_t>>(element_type);
+    }
+
+    case ElementTypeId::kInt8: {
+      return std::make_unique<NumericAbstractFlexVector<int8_t>>(element_type);
+    }
+
+    case ElementTypeId::kInt16: {
+      return std::make_unique<NumericAbstractFlexVector<int16_t>>(element_type);
+    }
+
+    case ElementTypeId::kInt32: {
+      return std::make_unique<NumericAbstractFlexVector<int32_t>>(element_type);
+    }
+
+    case ElementTypeId::kInt64: {
+      return std::make_unique<NumericAbstractFlexVector<int64_t>>(element_type);
+    }
+
+    case ElementTypeId::kFloat: {
+      return std::make_unique<NumericAbstractFlexVector<float>>(element_type);
+    }
+
+    case ElementTypeId::kDouble: {
+      return std::make_unique<NumericAbstractFlexVector<double>>(element_type);
+    }
+
+    case ElementTypeId::kBool: {
+      return std::make_unique<GenericAbstractFlexVector<bool>>(element_type);
+    }
+
+    case ElementTypeId::kString: {
+      return std::make_unique<GenericAbstractFlexVector<std::string>>(element_type);
+    }
+
+    case ElementTypeId::kTimestamp: {
+      return std::make_unique<GenericAbstractFlexVector<DateTime>>(element_type);
+    }
+
+    case ElementTypeId::kLocalDate: {
+      return std::make_unique<GenericAbstractFlexVector<LocalDate>>(element_type);
+    }
+
+    case ElementTypeId::kLocalTime: {
+      return std::make_unique<GenericAbstractFlexVector<LocalTime>>(element_type);
+    }
+
+    default: {
+      auto message = fmt::format("Programming error: elementTypeId {} not supported here",
```

Comment:
Suggest consider "Internal error" instead of "Programming error".

Reply:
Done.